app/services/notification_service.rb

```diff
@@ -0,0 +1,37 @@
+class NotificationService
+  def send_compatible_target(users, target)
+    params = {
+      data: { 'target_id': target.id },
+      filters: users_filter(users),
+      contents: { 'en': I18n.t('api.notification.target.new', title: target.title) }
+    }
+
+    delay.send_notification(params)
+  end
+
+  def send_test(email)
+    params = {
+      filters: [{ 'field': 'tag', 'key': 'email', 'relation': '=', 'value': email }],
+      contents: { 'en': 'This is a test' }
+    }
+
+    delay.send_notification(params)
+  end
+
+  private
+
+  def send_notification(params)
+    params[:app_id] = '08effaf6-443b-4a18-a096-ed472783e463'
+    OneSignal::OneSignal.api_key = 'YzRmNWQwNTQtODVjNC00Y2NhLWJhYTMtM2JmOGZhMTFiMWY0'
```

Comment:
Never put api keys inside code. These should not be visible for anyone. Add it to `application.yml`